Masterbar: adjust all icon styling

arcangelini opened the pull request:
## Changes proposed in this Pull Request

With the recent addition of the help center to the masterbar there is a discrepancy in icon styling and notification bubbles. Per our conversation p1650366069745269-slack-C02T4NVL4JJ, this aligns all the notification bubbles to the right.

## Testing instructions

Pull and `yarn start`
Get a notification by commenting on your own blog and add something to your cart.

| Mobile | Desktop |
| - | - |
| <img width="156" alt="Markup 2022-04-29 at 01 33 23" src="https://user-images.githubusercontent.com/33258733/165864769-22674287-eb22-4d19-99ec-fb22b01ce64c.png"> | <img width="261" alt="Markup 2022-04-29 at 01 33 05" src="https://user-images.githubusercontent.com/33258733/165864807-5d2e03b9-6d7f-4813-9bc9-b523bafa6268.png"> |

Closes #62756

arcangelini committed:
Masterbar: adjust all icon styling

poligilad-auto commented:
Thanks @arcangelini!
Really like seeing everything aligned like that.

This is kind of nit-picking, but the current icons seems just 1px to high. Try this one instead, I moved it a bit lower.

![bell-straight-2](https://user-images.githubusercontent.com/83961704/165044749-a082b008-7661-4283-aebd-df30120374a7.svg)

tyxla commented on client/layout/masterbar/bell-straight-icon.tsx:
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export const BellIcon = (): React.ReactNode => (
```
Do we need all of `React` just to use `ReactNode` type?

Could we import just the type instead?

```suggestion
import type { ReactNode } from 'react'

export const BellIcon = (): ReactNode => (
```

arcangelini replied:
Replaced this with function instead : )

tyxla commented on client/layout/masterbar/masterbar-cart/masterbar-cart-button.tsx:
```diff
@@ -64,9 +64,10 @@ export function MasterbarCartButton( {
 				ref={ cartButtonRef }
 			>
-				<MasterbarCartCount productsInCart={ responseCart.products.length } />
+				{ responseCart.products.length > 0 && <span className={ 'masterbar-cart__bubble' } /> }
```
No need to add extra expression evaluation when we know it's going to be a string for sure.

```suggestion
				{ responseCart.products.length > 0 && <span className="masterbar-cart__bubble" /> }
```

tyxla commented on client/layout/masterbar/notifications.jsx:
```diff
@@ -112,20 +113,15 @@ class MasterbarItemNotifications extends Component {
-							'notification-indicator-animation-state-' + Math.abs( this.state.animationState )
-						}
-					/>
+					{ this.state.newNote && <span className={ 'masterbar__notifications-bubble' } /> }
```
```suggestion
					{ this.state.newNote && <span className="masterbar__notifications-bubble" /> }
```

tyxla reviewed (commented):
Leaving the design aspect for other folks to review, I just left a few drive-by nits as I was glancing through the code.

arcangelini committed:
Replace icons with bubble integrated version

arcangelini committed:
Remove unused code and restructure

arcangelini committed:
Cleanup cart CSS

arcangelini commented:
Updated to use the new icons given by @poligilad-auto

arcangelini committed:
Move all styles for notifications into separate CSS

arcangelini committed:
Move all styles for notifications into separate CSS

alshakero commented on client/layout/masterbar/masterbar-cart/masterbar-cart-button-style.scss:
```diff
@@ -1,58 +1,6 @@
-		font-size: 0.6rem; /* stylelint-disable-line scales/font-sizes */
-		padding: 1px 4px;
-		top: 4px;
-	}
```
![image](https://user-images.githubusercontent.com/17054134/165588934-987ad5c6-100b-434a-a730-303cb52b38e2.png)

alshakero commented on client/layout/masterbar/masterbar-cart/masterbar-cart-icon.tsx:
```diff
@@ -0,0 +1,34 @@
+				<path
+					id="bubble"
+					fill="var( --color-masterbar-unread-dot-background )"
+					stroke={
+						active
+							? 'var( --color-masterbar-item-active-background )'
+							: 'var( --color-masterbar-background )'
+					}
+					strokeWidth="2px"
```
Stroke, and the CSS for `--color-masterbar-item-active-background` can go. New icons are proper transparent!
```suggestion
```

alshakero commented on client/layout/masterbar/masterbar-notifications/notifications-bell-icon.tsx:
```diff
@@ -0,0 +1,34 @@
+
+export const BellIcon: React.FC< Props > = ( { newItems, active } ) => (
+	<svg xmlns="http://www.w3.org/2000/svg" width="26" height="25" viewBox="0 0 24 24">
+		<g id="without-bubble">
+			<path
+				fill="var( --color-masterbar-text )"
+				d="M9.9,20h4c0,0.5-0.2,1-0.6,1.4   c-0.8,0.8-2,0.8-2.8,0C10.1,21,9.9,20.5,9.9,20z M20,17.5v1H4v-1l0.9-0.7C5.5,16.3,6,15.5,6,15l0-5.5c0-3.3,2.7-6,6-6   c3.3,0,6,2.7,6,6V15c0,0.5,0.5,1.4,1.1,1.8L20,17.5z"
+			/>
+		</g>
```
Same as above.

alshakero commented on client/layout/masterbar/masterbar-cart/masterbar-cart-icon.tsx:
```diff
@@ -0,0 +1,34 @@
+
+export const CartIcon: React.FC< Props > = ( { newItems, active } ) => (
+	<svg xmlns="http://www.w3.org/2000/svg" width="26" height="25" viewBox="0 0 24 24">
+		<g id="without-bubble">
+			<path
+				fill="var( --color-masterbar-text )"
+				d="M9,20c0,1.1-0.9,2-2,2s-2-0.9-2-2s0.9-2,2-2   S9,18.9,9,20z M17,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S18.1,18,17,18z M17.4,13c0.9,0,1.7-0.7,2-1.6L21,5H7V4c0-1.1-0.9-2-2-2   H3v2h2v1v8v2c0,1.1,0.9,2,2,2h12c0-1.1-0.9-2-2-2H7v-2H17.4z"
+			/>
+		</g>
+		{ newItems && (
+			<g id="with-bubble">
```
you only need stroke below because the `without-bubble` cart is unchipped, and it's visible.
```suggestion
		{!newItems ? <g id="without-bubble">
			<path
				fill="var( --color-masterbar-text )"
				d="M9,20c0,1.1-0.9,2-2,2s-2-0.9-2-2s0.9-2,2-2   S9,18.9,9,20z M17,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S18.1,18,17,18z M17.4,13c0.9,0,1.7-0.7,2-1.6L21,5H7V4c0-1.1-0.9-2-2-2   H3v2h2v1v8v2c0,1.1,0.9,2,2,2h12c0-1.1-0.9-2-2-2H7v-2H17.4z"
			/>
		</g> : 
			<g id="with-bubble">
```

alshakero commented on client/layout/masterbar/masterbar-notifications/notifications-bell-icon.tsx:
```diff
@@ -0,0 +1,34 @@
+				<path
+					id="bubble"
+					fill="var( --color-masterbar-unread-dot-background )"
+					stroke={
+						active
+							? 'var( --color-masterbar-item-active-background )'
+							: 'var( --color-masterbar-background )'
+					}
+					strokeWidth="2px"
```
```suggestion
```

alshakero reviewed (changes requested):
So freaking refreshing! Left some comments. And all ids can go, they're meant just to help you during development.

poligilad-auto commented:
Hey @arcangelini, looking good.

A few small things:
- The icon sizes should be 24x24px. Now I see the bell as 26x25px.
- For some reason the bubble is appearing smaller than in the SVG... 

<img width="109" alt="image" src="https://user-images.githubusercontent.com/83961704/165697202-7d604a41-404f-4c8d-80d5-7e46384c43c1.png">
Screenshot on the left | SVG on the right

arcangelini committed:
Remove stroke and switch order

arcangelini commented:
@poligilad-auto Haha sorry about that I misread the SVG and added a stroke which caused that. Everything should be updated including the screenshots above.

One thing I tested out but I didn't know if you wanted. I moved the icons to both be on the left of the profile rather than splitting it. This is how it is in mobile (because the profile icon is in the hamburger menu) and it looked to fit a little better. I can switch it back if this is not how you envisioned it. Let me know!

poligilad-auto commented:
Cool, the dots look good now 👍

> One thing I tested out but I didn't know if you wanted. I moved the icons to both be on the left of the profile rather than splitting it. This is how it is in mobile (because the profile icon is in the hamburger menu) and it looked to fit a little better. I can switch it back if this is not how you envisioned it. Let me know!

Understand the thinking, but please move the icons back, it's important that the notification icon is on the most right side on desktop.

Also, I just saw this small misalignment. Maybe it's caused by these changes?
<img width="431" alt="image" src="https://user-images.githubusercontent.com/83961704/165898128-cf0543b3-7be8-4d88-8999-bb6e7c7051ec.png">

arcangelini committed:
Move icons back

alshakero commented on client/layout/masterbar/masterbar-cart/masterbar-cart-icon.tsx:
```diff
@@ -0,0 +1,28 @@
+					fill="var( --color-masterbar-text )"
+					d="M14.1,5c0.5,2.7,2.8,4.9,5.6,5l-0.4,1.4   c-0.2,0.9-1,1.6-2,1.6H7v2h10c1.1,0,2,0.9,2,2H7c-1.1,0-2-0.9-2-2v-2V5V4H3V2h2c1.1,0,2,0.9,2,2v1H14.1z M7,22c1.1,0,2-0.9,2-2   s-0.9-2-2-2s-2,0.9-2,2S5.9,22,7,22z M15,20c0-1.1,0.9-2,2-2s2,0.9,2,2s-0.9,2-2,2S15,21.1,15,20z"
+				/>
+				<path
+					fill="var( --color-masterbar-unread-dot-background )"
+					d="M20,8c2.2,0,4-1.8,4-4s-1.8-4-4-4s-4,1.8-4,4S17.8,8,20,8z"
+				/>
```
Nit (this is an Illustrator issue): this can be just a circle, doesn't need to be path.

```suggestion
				<circle cx="20" cy="4" r="4" fill="var( --color-masterbar-unread-dot-background )"/>
```

alshakero commented on client/layout/masterbar/masterbar-notifications/notifications-bell-icon.tsx:
```diff
@@ -0,0 +1,28 @@
+					fill="var( --color-masterbar-text )"
+					d="M18,9.7V15c0,0.5,0.5,1.4,1.1,1.8l0.9,0.7v1H4v-1l0.9-0.7C5.5,16.3,6,15.5,6,15l0-5.5   c0-3.3,2.7-6,6-6c0.7,0,1.4,0.1,2,0.3c0,0.1,0,0.1,0,0.2C14,6.6,15.7,8.8,18,9.7z M13.9,20h-4c0,0.5,0.2,1,0.6,1.4   c0.8,0.8,2,0.8,2.8,0C13.8,21,13.9,20.5,13.9,20z"
+				/>
+				<path
+					fill="var( --color-masterbar-unread-dot-background )"
+					d="M20,8c2.2,0,4-1.8,4-4s-1.8-4-4-4s-4,1.8-4,4S17.8,8,20,8z"
+				/>
```
```suggestion
				<circle cx="20" cy="4" r="4" fill="var( --color-masterbar-unread-dot-background )"/>
```

alshakero commented on client/layout/masterbar/style.scss:
```diff
@@ -633,93 +566,6 @@ $masterbar-color-secondary: #101517;
-		transform: translateX( 1px );
-
-		// Animation
-		animation: bubble-unread-indication 0.5s linear both;
```
Removing this animation is a regression. We'll need to keep it. You just have to adjust the `transform-origin`. Probably to `20px 4px` (it's relative to the SVG's viewbox), please ping me if you need anything SVG, I love SVGs.

alshakero commented on client/layout/masterbar/style.scss:
```diff
@@ -633,93 +566,6 @@ $masterbar-color-secondary: #101517;
-	}
-}
-
-@keyframes bubble-unread-indication {
-	30% {
-		transform: scale3d( 1.5, 1.5, 1.5 );
-	}
-	60% {
-		transform: scale3d( 0.85, 0.85, 0.85 );
-	}
-	80% {
-		transform: scale3d( 1.15, 1.15, 1.15 );
-	}
-}
```
The `bubble-unread-indication` should stay.

alshakero commented on client/layout/masterbar/masterbar-notifications/notifications-button.jsx:
```diff
@@ -112,22 +115,14 @@ class MasterbarItemNotifications extends Component {
-					{ this.props.children }
-					<span
-						className="masterbar__notifications-bubble"
-						key={
-							'notification-indicator-animation-state-' + Math.abs( this.state.animationState )
-						}
```
This key prop should be set to the `BellIcon`. Changing the key causes re-rendering of the element, thus running the animation again. This is a nice way to show the bubble dancing when you get a new notification.

arcangelini replied:
This was all removed including the `span`. Are you suggesting adding this key to the `MasterbarItem`?

alshakero commented on client/layout/masterbar/masterbar-notifications/notifications-style.scss:
```diff
@@ -0,0 +1,39 @@
+		}
+	}
+
+	.gridicon {
+		fill: var( --color-text-inverted );
+		margin-top: 1px;
+
+		@media only screen and ( max-width: 782px ) {
+			flex-shrink: 0;
+		}
+	}
+
+	.gridicon + .masterbar__item-content {
+		padding: 0;
+
+		@include breakpoint-deprecated( '<480px' ) {
+			display: block;
+		}
+	}
```
From what I see, notifications item doesn't have `gridicons` anymore. This can probably be deleted.

alshakero reviewed (changes requested):
So neat and smooth. Even the bubble looks smoother than before, probably due to anti-aliasing.

arcangelini committed:
Minor tweaks - replaced path with circle

arcangelini committed:
Hookup animations

arcangelini committed:
Bigger animation

arcangelini committed:
Make the dot throb WITH outline

arcangelini committed:
Fix dot hover

alshakero reviewed (approved):
Such an upgrade ✨
